bot/modals/class_modal.py

```diff
@@ -0,0 +1,192 @@
+from typing import Union
+
+import discord
+from discord import Interaction, TextStyle
+from discord.ui import Modal, TextInput
+
+from bot.consts import Colors
+from bot.data.class_repository import ClassRepository
+from bot.messaging.events import Events
+from bot.models.class_models import ClassChannelScaffold
+from bot.sock_bot import SockBot
+from bot.utils.helpers import error_embed
+
+# Min and max of major length, i.e., 'CPSC', 'HCC', 'MATH'
+MIN_MAJOR_LEN = 3
+MAX_MAJOR_LEN = 4
+# Min and max of class title length, i.e., 'Data Structures and Algorithms', etc.
+MIN_TITLE_LEN = 5
+MAX_TITLE_LEN = 50
+# Min and max of class instructor, i.e., 'Dean', 'Widman', 'VanScoy', etc.
+MIN_INSTR_LEN = 2
+MAX_INSTR_LEN = 15
+# Max length of class description, i.e., 'Students learn about...'
+MAX_DESCR_LEN = 500
+# Min and max of class number, i.e., 1060, 4910, 8400, etc.
+MIN_CLASS_NUM = 1000
+MAX_CLASS_NUM = 8999
+
+
+class AddClassModal(Modal):
```

Comment:
This needs to implement on_error() so that we can log the errors in the error channel

Reply:
probably need to pass the error callback through the modal object creation in manage_classes_cog

Reply:
<img width="891" alt="image" src="https://user-images.githubusercontent.com/25359755/221428327-9e4f22fa-d857-48a1-9ee2-46c9e3fa92a0.png">